Clarify usage of "otel." attribute namespace (#1640)
open-telemetry/opentelemetry-specification: specification/common/attribute-and-label-naming.md

(more precisely to
[U+0021 .. U+007E](https://en.wikipedia.org/wiki/Basic_Latin_(Unicode_block)#Table_of_characters)
subset of Unicode code points).

## otel.* Namespace

> Attribute names that start with `otel.` are reserved to be defined by

Reviewer: ```suggestion
Attribute and label names that start with `otel.` are reserved to be defined by
```

OpenTelemetry specification. These are typically used to express OpenTelemetry
concepts in formats that don't have a corresponding concept.

> For example `otel.library.name` attribute is used to record the instrumentation

Reviewer: ```suggestion
For example, the `otel.library.name` attribute is used to record the instrumentation
```

> For example `otel.library.name` attribute is used to record the instrumentation

Reviewer 2: Nitpick: I wonder if you can call that an attribute. This string will only appear in the Jager/Zipkin exporters at which point its debatable if this is an attribute or another kind of key/value pair.

Author: It is supposed to be recorded in the same map (tags in Jaeger vocabulary) where we record attributes. So, it occupies the same namespace of attribute names. I am not sure if there is a better term for this, I am open to suggestions.

library name, which is an OpenTelemetry concept that is natively represented in
OTLP, but does not have an equivalent in other telemetry formats and protocols.

> Any additions to `otel.*` namespace should be approved as part of OpenTelemetry

Reviewer: ```suggestion
Any additions to the `otel.*` namespace should be approved as part of OpenTelemetry
```

Reviewer 2: ```suggestion
Any additions to the `otel.*` namespace MUST be approved as part of OpenTelemetry
```
